static/js/message_list_view.js

```diff
@@ -57,22 +57,53 @@ function same_recipient(a, b) {
     return util.same_recipient(a.msg, b.msg);
 }
 
-function message_was_only_moved(message) {
-    // Returns true if the message has had its stream/topic edited
-    // (i.e. the message was moved), but its content has not been
-    // edited.
+function analyze_edit_history(message) {
+    // Returns a dict of booleans that describe the message's history:
+    //   * edited: if the message has had its content edited
+    //   * moved: if the message has had its stream/topic edited
+    //   * resolve_toggled: if the message has had a topic resolve/unresolve edit
+    let edited = false;
     let moved = false;
+    let resolve_toggled = false;
+
     if (message.edit_history !== undefined) {
         for (const edit_history_event of message.edit_history) {
             if (edit_history_event.prev_content) {
-                return false;
+                edited = true;
+            }
+
+            if (edit_history_event.prev_stream) {
+                moved = true;
             }
-            if (edit_history_event.prev_topic || edit_history_event.prev_stream) {
+
+            if (edit_history_event.prev_topic) {
+                // We know it has a topic edit. Now we need to determine if
+                // it was a true move or a resolve/unresolve.
+                if (
+                    edit_history_event.topic.startsWith(resolved_topic.RESOLVED_TOPIC_PREFIX) &&
+                    edit_history_event.topic.slice(2) === edit_history_event.prev_topic
```

Comment:
I think these can use the new in the last week `static/shared/js/resolved_topics` library for these checks, and we should probably do that to avoid an abstraction violation.

Reply:
Yes, I should have looked more carefully at that when I did the rebase yesterday.